### PR TITLE
Bug/gh739 aligning column headers

### PR DESCRIPTION
**Pull request by omaratrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Align column headers in element definition tree, product tree and requirements tree.
image

image

image

### CODE REVIEW

File: EngineeringModel/Views/ElementDefinitionBrowser/ElementDefinitionsBrowser.xaml [hunk 1]
```diff
@@ -225,7 +98,10 @@
                                       NavigationStyle="Cell"
                                       ShowHorizontalLines="False"
                                       ShowIndicator="False"
-                                      ShowNodeImages="False"
+                                      RowIndent="22"
+                                      ShowNodeImages="True"
+                                      NodeImageSelector="{StaticResource ElementDefinitionTreeListNodeSelector}"
+                                      NodeImageSize="200,16"
```
Comment (alexatrhea) [inline EngineeringModel/Views/ElementDefinitionBrowser/ElementDefinitionsBrowser.xaml:104]: forgot to set this 200 back to 22 I think. Please check whether 22 is correct.

File: ProductTree/Views/ProductTree.xaml [hunk 2]
```diff
@@ -184,7 +134,8 @@
                                   NavigationStyle="Cell"
                                   ShowHorizontalLines="False"
                                   ShowIndicator="False"
-                                  ShowNodeImages="False"
+                                  ShowNodeImages="True"
```
Comment (alexatrhea) [inline ProductTree/Views/ProductTree.xaml:137]: Didnt have to add here `NodeImageSize="20,16"` ?

**Review by alexatrhea — CHANGES_REQUESTED** [review]

**Comment by alexatrhea** [comment]
By the way @lxatrhea @samatrhea the consequence of this PR is that we cannot put the overlays to the left side of the icon any more. This is because of the way node images are aligned, meant that if a left side overlay was present the the icons would get misaligned against ones that did not have an overlay.

File: ProductTree/Views/ProductTree.xaml (hunk 2 above)
Comment (omaratrhea) [reply ProductTree/Views/ProductTree.xaml:137]: Not needed here

**Review by lxatrhea — APPROVED** [review]

File: ProductTree/Selectors/ProductTreeListNodeSelector.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        /// </summary>
+        /// <param name="rowData"><see cref="TreeListRowData"/></param>
+        /// <returns>ImageSource</returns>
+        public override ImageSource Select(TreeListRowData rowData)
+        {
+            var thingStatus = ((IHaveThingStatus)rowData.Row).ThingStatus;
+
+            if (rowData.Row is ParameterOrOverrideBaseRowViewModel parameterOrOverrideBaseRow)
+            {
+                var productTreeIconConverter = new ProductTreeIconUriConverter();
```
Comment (samatrhea) [inline ProductTree/Selectors/ProductTreeListNodeSelector.cs:55]: do we still need the `ProductTreeIconUriConverter` now that you have change the design pattern?

**Review by samatrhea — CHANGES_REQUESTED** [review]

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**